core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -164,6 +169,20 @@ public void process(final Archive<?> applicationArchive, final TestClass testCla
         tryToAdd(libs, libraryContainer, "WEB-INF/lib/scala-library-", "cucumber.api.scala.ScalaDsl", "scala.App");
     }
 
+    private boolean archiveContains(Archive<?> archive, Class<?> clazz) {

```

Comment:
nice trick to use the test class as marker!

Side note: maybe just use get instead of iterating on all nodes trying:

    classname.replace('.', '/') + ".class"
    WEB-INF/classes/classname.replace('.', '/') + ".class"

?